src/main/java/com/puppycrawl/tools/checkstyle/TreeWalker.java

```diff
@@ -232,14 +232,15 @@ protected void processFiltered(File file, FileText fileText) throws CheckstyleEx
      * Returns filtered set of {@link LocalizedMessage}.
      * @param fileName path to the file
      * @param fileContents the contents of the file
+     * @param rootAST root AST element {@link DetailAST} of the file
      * @return filtered set of messages
      */
-    private SortedSet<LocalizedMessage> getFilteredMessages(String fileName,
-                                                            FileContents fileContents) {
+    private SortedSet<LocalizedMessage> getFilteredMessages(
+            String fileName, FileContents fileContents, DetailAST rootAST) {
         final SortedSet<LocalizedMessage> result = new TreeSet<>(messages);
         for (LocalizedMessage element : messages) {
             final TreeWalkerAuditEvent event =
-                    new TreeWalkerAuditEvent(fileContents, fileName, element);
+                    new TreeWalkerAuditEvent(fileContents, fileName, element, rootAST);
             for (TreeWalkerFilter filter : filters) {
```

Comment:
We do not need to loop over the set of localized messages in canse filters set is empty. Add the check.